packages/vertexai/src/requests/schema-builder.ts

```diff
@@ -0,0 +1,197 @@
+import { SchemaInterface, SchemaType, SchemaParams } from '../types/schema';
+
+export abstract class Schema implements SchemaInterface {
+  /**
+   * Optional. The type of the property. {@link
+   * SchemaType}.
+   */
+  type: SchemaType;
+  /** Optional. The format of the property.
+   * Supported formats:
+   *  for NUMBER type: "float", "double"
+   *  for INTEGER type: "int32", "int64"
+   *  for STRING type: "email", "byte", etc
+   */
+  format?: string;
+  /** Optional. The description of the property. */
+  description?: string;
+  /** Optional. Whether the property is nullable. Defaults to false. */
+  nullable: boolean;
```

Comment:
FYI this isn't a JS concept of nullable, it's you giving a schema to the model to fill in (whether you're asking it to do function calling or give you a JSON response), and you're telling the model whether it's ok to return this field to you with a value of null. I don't think non JS languages have undefined so I think it's just null. So if you say "required: true" and "nullable: false" that means, you have to fill out this field when you send it back to me, and it can't be null. "required: true"/"nullable: true" I think means, "this field needs to be in the JSON and it needs to be filled with some actual value". Actually JSON is JS I guess, but the official spec says "undefined" isn't a valid JSON value, just null.